acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/publickeys/YubiHsmKeysAcceptanceTest.java

```diff
@@ -0,0 +1,145 @@
+          "4b6b5c682f2db7e510e0c00ed67ac896c21b847acadd8df29cf63a77470989d2",
+          "13086d684f4b1a1632178a8c5be08a2fb01287c4a78313c41373701eb8e66232",
+          "25296867ee96fa5b275af1b72f699efcb61586565d4c3c7e41f4b3e692471abd",
+          "10e1a313e573d96abe701d8848742cf88166dd2ded38ac22267a05d1d62baf71",
+          "0bdeebbad8f9b240192635c42f40f2d02ee524c5a3fe8cda53fb4897b08c66fe",
+          "5e8d5667ce78982a07242739ab03dc63c91e830c80a5b6adca777e3f216a405d");
+
+  // following SECP keys are expected to be pre-loaded in YubiHSM as Opaque Data (HEX informat).
+  // Opaque Id 20, 21, 22.
+  private static final List<String> PRE_LOADED_SECP_PRIVATE_KEYS =
+      List.of(
+          "ee0b33ddc9d584a567b0df55e88550966f6b830cb9dff6334c01d416bc129ef3",
+          "c19744babeaf9fb0fdbb0cc0076472f5ce3f23551d5c2cd0145b8e341135156c",
+          "188523561ef3688b43a96ecb110583a737e6067194a5c9f5570caa9c122ad8ef");
+
+  // Using usb connector
+  private static final String CONNECTOR_URL = "yhusb://";
+  private static final String ADDITIONAL_INIT_CONFIG = "debug libdebug";
+
+  private static final short AUTH_ID = 1;
+  private static final String PASSWORD = "password";
+
+  private static final MetadataFileHelpers METADATA_FILE_HELPERS = new MetadataFileHelpers();
+
+  @BeforeEach
+  void checkPkcs11ModuleExist() {
+    assertThat(Path.of(MODULE_PATH)).exists();
+  }
+
+  @Test
+  public void blsKeysAreLoadedFromYubiHsm() {
+    createBlsConfigurationFiles();
+
+    initAndStartSigner(calculateMode(KeyType.BLS));
+
+    final Response response = signer.callApiPublicKeys(KeyType.BLS);
+    final String[] expectedPublicKeys =
+        PRE_LOADED_BLS_PRIVATE_KEYS.stream()
+            .map(key -> BLSSecretKey.fromBytes(Bytes32.fromHexString(key)).toPublicKey().toString())
+            .toArray(String[]::new);
+    validateApiResponse(response, containsInAnyOrder(expectedPublicKeys));
+  }
+
+  private void createBlsConfigurationFiles() {
+    for (short opaqueDataId = (short) 1;
+        opaqueDataId <= PRE_LOADED_BLS_PRIVATE_KEYS.size();
+        opaqueDataId++) {
+      final Path configFile = testDirectory.resolve("yubihsm_" + opaqueDataId + ".yaml");
+      METADATA_FILE_HELPERS.createYubihsmYamlFileAt(
+          configFile,
+          MODULE_PATH,
+          CONNECTOR_URL,
+          ADDITIONAL_INIT_CONFIG,
+          AUTH_ID,
+          PASSWORD,
+          opaqueDataId,
+          KeyType.BLS);
+    }
+  }
+
+  @Test
+  public void secpKeysAreLoadedFromYubiHsm() {
+    createSecpConfigurationFiles();
+
+    initAndStartSigner(calculateMode(KeyType.SECP256K1));
+
+    final Response response = signer.callApiPublicKeys(KeyType.SECP256K1);
+
+    final String[] expectedPublicKeys =
+        PRE_LOADED_SECP_PRIVATE_KEYS.stream().map(this::getPublicKey).toArray(String[]::new);
+    validateApiResponse(response, containsInAnyOrder(expectedPublicKeys));
+  }
+
+  private void createSecpConfigurationFiles() {
+    for (short opaqueDataId = 20;
```

Comment:
whats the magic 20?

Reply:
expected starting OpaqueObjId for SECP keys. I'll create a constant.